homework/index.js

```diff
@@ -29,18 +33,218 @@
+    const selectList = document.getElementById('repo-select');
+    repositoriesData
+      .sort((a, b) => a.name.localeCompare(b.name))
+      .forEach((repository, index) => {
+        createAndAppend('option', selectList, {
+          text: repository.name,
+          value: index,
+        });
+      });
+    // create an eventlistener for select list options
+    selectList.addEventListener('change', () => {
+      selectList.parentElement.nextElementSibling.innerHTML = '';
+      cb(repositoriesData[selectList.value], repositoriesData[selectList.value].contributors_url);
+    });
+  };
+  // fetch repositories and contributions data as a callback hell so that once repositories data is fetched first, then the contribution data will be fetched.
+  const fetchRepositoriesAndContributions = url => {
+    fetchJSON(url, (errRepository, repositoriesData) => {
+      if (errRepository) {
+        createAndAppend('div', root, {
+          text: errRepository.message,
+          class: 'alert alert-danger display-3 text-center',
+        });
         return;
       }
-      createAndAppend('pre', root, { text: JSON.stringify(repositories, null, 2) });
+      appendRepositoriesToSelect(repositoriesData, (repositoryObj, contributionsURL) => {
+        createRepositoryLayout(repositoryObj);
+        fetchJSON(contributionsURL, (errContributions, contributionsObj) => {
+          if (errContributions) {
+            createAndAppend('div', root, {
+              text: `${errContributions.message}... The data cannot be fetched`,
+              class: 'alert alert-danger display-3 text-center',
+            });
+            return;
+          }
+          createContributionsLayout(contributionsObj);
+        });
+      });
+    });
+  };
+  // create page layout under the root div
+  const buildWebPage = url => {
+    const header = createAndAppend('header', root, {
+      class: 'container jumbotron mt-3 mb-3 px-5',
     });
+    createAndAppend('main', root, { id: 'main', class: 'container align-items-start' });
+    const headerDiv = createAndAppend('div', header, {
+      class: 'header-group d-flex flex-row justify-content-between mb-5 px-5',
+    });
+    createAndAppend('h2', headerDiv, {
+      text: 'Hack Your Future Repositories',
+      class: 'display-4 d-flex align-items-center mx-auto col-sm header-title',
+    });
+    const hyfLink = createAndAppend('a', headerDiv, {
+      href: 'https://www.hackyourfuture.net/',
+      target: '_blank',
+    });
+    createAndAppend('img', hyfLink, {
+      class: 'img-fluid rounded img-thumbnail mx-auto',
+      src: './hyf.png',
+      alt: 'hack your future thumbnail',
+    });
+    const selectList = createAndAppend('select', header, {
+      id: 'repo-select',
+      class: 'form-control',
+      'aria-label': 'Hack Your Future Repositories Selection',
+    });
+    // Create Select Box and its options
+    createAndAppend('option', selectList, {
```

Comment:
We used this in class (a dummy first option), but the homework assignment asks for you to select the first repo from the sorted list by default and display its details.

Reply:
I have tried to get the dummy values for the first option, It is easy to get the repository value, but in my structure i haven't found a minor solution for the contributors dummy value because i wrote in the beginning only if get data when there is any change in the select-list.  Can you help me to find a solution?

Reply:
The `change` event for the `<select>` element is only fired when you actually change the selection. This event is not fired when your application loads. So you need to call the same functions that your event handler calls manually, using the data for the first repo in the list.

Reply:
Ok. I found now. Thank you.

Reply:
Logically speaking, there should be a one-to-one correspondence between the current option displayed in the `<select>` box and the currently displayed repo details. So either you have a placeholder option which says **Select a Repository** without displaying any repo details, or you display the first repo name in the `<select>` box along with the corresponding repo details in the page.

In other words, you should now remove this placeholder option so that the `<select>` box displays the name of the first repo in the list (**alumni**).

Reply:
I have removed the placeholder and pushed again . Thank you for your time, J

Reply:
I corrected es lint errors. and pushed again. Thank you for all.